wings/market/coinbase_pro_market.pyx

```diff
@@ -0,0 +1,816 @@
+        in_flight_deposits = [d for d in self._in_flight_deposits.values() if not d.has_tx_receipt]
+        tx_hash_to_deposit_map = dict((d.tx_hash, d) for d in in_flight_deposits)
+
+        for d in in_flight_deposits:
+            receipt = self._w3.eth.getTransactionReceipt(d.tx_hash)
+            if receipt is None or receipt.blockHash is None:
+                continue
+            if receipt.status == 0:
+                d.has_tx_receipt = True
+                self.c_did_fail_tx(d.tracking_id)
+            if receipt.status == 1:
+                self.logger().info(f"Received {d.amount} {d.currency} from {d.from_address} via tx hash {d.tx_hash}.")
+                self.c_trigger_event(self.MARKET_RECEIVED_ASSET_EVENT_TAG,
+                                     MarketReceivedAssetEvent(
+                                         self._current_timestamp,
+                                         d.tracking_id,
+                                         d.from_address,
+                                         d.to_address,
+                                         d.currency,
+                                         float(d.amount)
+                                     ))
+                d.has_tx_receipt = True
+                self.c_stop_tracking_deposit(d.tracking_id)
+
+    async def _update_trading_rules(self):
+        cdef:
+            # The poll interval for withdraw rules is 60 seconds.
+            int64_t last_tick = <int64_t>(self._last_timestamp / 60.0)
+            int64_t current_tick = <int64_t>(self._current_timestamp / 60.0)
+        if current_tick > last_tick or len(self._trading_rules) <= 0:
+            product_info = await self._api_request('get', path_url="/products")
+            trading_rules_list = TradingRule.parse_exchange_info(product_info)
+            self._trading_rules.clear()
+            for trading_rule in trading_rules_list:
+                self._trading_rules[trading_rule.symbol] = trading_rule
+
+    async def _update_order_status(self):
+        cdef:
+            int64_t last_tick = <int64_t>(self._last_timestamp / 10.0)
+            int64_t current_tick = <int64_t>(self._current_timestamp / 10.0)
+
+        if len(self._in_flight_orders) == 0:
+            return
+
+        tracked_orders = list(self._in_flight_orders.values())
+        results = await self.list_orders()
+        order_dict = dict((result["id"], result) for result in results)
+
+        for tracked_order in tracked_orders:
+            exchange_order_id = await tracked_order.get_exchange_order_id()
+            order_update = order_dict.get(exchange_order_id)
+            if order_update is None:
+                self.logger().error(f"Error fetching status update for the order {tracked_order.client_order_id}: "
+                                    f"{order_update}.")
+                continue
+
+            # Calculate the newly executed amount for this update.
+            previous_is_done = tracked_order.is_done
```

Comment:
we dont need this logic if we are deleting orders right away, not using expiry queue

Reply:
fixed